syncany-lib/src/main/java/org/syncany/operations/up/Indexer.java

```diff
@@ -367,11 +330,21 @@ private void addFileVersion(FileProperties fileProperties) {
 			PartialFileHistory lastFileHistory = guessLastFileHistory(fileProperties);
 			FileVersion lastFileVersion = (lastFileHistory != null) ? lastFileHistory.getLastVersion() : null;
 
-			// 2. Create new file history/version
+			// 2. If file type changed, "close" the old file history by adding a file version that deletes the old file/directory
+			PartialFileHistory deletedFileHistory = null;
```

Comment:
This seems to make sense, but it is not related to the memory issues, is it? Did you discover an issue here?